### PR TITLE
chore: Update cli parameters with always fail open

### PR DESCRIPTION
**Pull request by annikalee**
Adds a flag to the cli config to always fail open and suppress errors when set

- [X] I ran `make setup && make` to update the generated code after editing a `.atd` file (TODO: have a CI check)
- [X] I made sure we're still backward compatible with old versions of the CLI.
      For example, the Semgrep backend need to still be able to *consume* data generated
	  by Semgrep 1.17.0.
      See https://atd.readthedocs.io/en/latest/atdgen-tutorial.html#smooth-protocol-upgrades

### CODE REVIEW

File: semgrep_output_v1.atd [hunk 1]
```diff
@@ -1472,6 +1472,8 @@ type engine_configuration = {
     ~generic_slow_rollout: bool;
     (* from 1.63.0 *)
     ?historical_config: historical_configuration option;
+    (* for customers to use the app to configure fail open options *)
```
Comment (salolivares) [inline semgrep_output_v1.atd:1475]:
```suggestion
    (* from 1.93 *)
    (* for customers to use the app to configure fail open options *)
```

Let's add the the version!

File: semgrep_output_v1.atd [hunk 2]
```diff
@@ -1472,6 +1472,8 @@ type engine_configuration = {
...
+    ~always_fail_open: bool;
```
Comment (salolivares) [inline semgrep_output_v1.atd:1476]: I'm actually unclear on this, perhaps @emjin can chime in. Is there a difference between `~` and `?` ?. I can't really find anything in the docs other than https://atd.readthedocs.io/en/latest/atdgen-tutorial.html#smooth-protocol-upgrades. 

Seems like this should be a safe upgrade and shouldn't break an existing scans when merged in (looks like we default to false if it's not defined).

Comment (emjin) [reply semgrep_output_v1.atd:1477]: `~` is a field with a default value, `?` is an optional field that can be None or Some x

File: semgrep_output_v1.atd [hunk 3]
```diff
@@ -1472,6 +1472,9 @@ type engine_configuration = {
     ~generic_slow_rollout: bool;
     (* from 1.63.0 *)
     ?historical_config: historical_configuration option;
+    (* from 1.93 *)
+    (* for customers to use the app to configure fail open options *)
```
Comment (emjin) [inline semgrep_output_v1.atd:1476]:
```suggestion
    (* sent by the app to indicate that fail open should always be enabled, overriding the CLI flag. coupling: server/semgrep_app/saas/models/deployment_products_mixin.py *)
```

Comment (emjin) [reply semgrep_output_v1.atd:1476]: Clarified what this flag does and added a pointer to the comment in semgrep-app

**Review by emjin — APPROVED** [review]

**Comment by annikalee** [comment]
@emjin I updated it to always_suppress_errors so asked for a re-review to make sure you're good with the new changes!

**Pull request merged**